packages/amplify-graphql-model-transformer/src/__tests__/model-transformer.test.ts

```diff
@@ -1906,4 +1906,239 @@ describe('ModelTransformer:', () => {
+        );
+      });
+
+      it('should include autoId when id is type String', async () => {
+        const schema = `
+          type Post @model {
+              id: String
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer()],
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should include autoId when id is type Int, Float, or Boolean', async () => {
+        const schema = `
+          type Foo @model {
+              id: Int
+              title: String!
+          }
+
+          type Bar @model {
+              id: Float
+              title: String!
+          }
+
+          type Baz @model {
+              id: Boolean
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer()],
+        });
+        expect(out.resolvers['Mutation.createFoo.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+        expect(out.resolvers['Mutation.createBar.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+        expect(out.resolvers['Mutation.createBaz.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should not include autoId when using a custom primary key', async () => {
+        const schema = `
+          type Post @model {
+              postId: ID! @primaryKey
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer(), new PrimaryKeyTransformer()],
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+    });
+
+    describe('sql', () => {
+      it('should include autoId for basic ID', async () => {
+        const schema = `
+          type Post @model {
+              id: ID!
+              postId: ID! @primaryKey
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer(), new PrimaryKeyTransformer()],
+          dataSourceStrategies: constructDataSourceStrategies(schema, makeStrategy(MYSQL_DB_TYPE)),
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should not include autoId for implicit ID', async () => {
```

Comment:
This isn't an implicit id, this is an explicit, right? I would not expect an `id` field to be generated here

Reply:
Fixed the name. This test is for no `id` field now.